onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/gelu_fusion.h"
+#include "core/graph/graph_utils.h"
+#include <deque>
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+static void IsInputConstant(const Graph& graph, const Node& node, std::tuple<bool, NodeArg*, NodeArg*>& ret) {
+  const auto& inputs = node.InputDefs();
+  bool found_constant = false;
+  NodeArg* gelu_non_const_input = nullptr;
+  NodeArg* gelu_const_input = nullptr;
+  for (auto& i : inputs) {
+    if (graph_utils::NodeArgIsConstant(graph, *i)) {
+      // Todo: check the constant for example be sqrt(2.0) or 1
+      found_constant = true;
+      gelu_const_input = const_cast<NodeArg*>(i);
+    } else {
+      gelu_non_const_input = const_cast<NodeArg*>(i);
+    }
+  }
+
+  ret = std::make_tuple(found_constant, gelu_non_const_input, gelu_const_input);
```

Comment:
>found_constant [](start = 24, length = 14)

Given gelu_const_input will be nullptr if found_constant is false, what value does found_constant add?

Reply:
I changed the way doing the check. use CheckConstantInput for check specified input_arg is constant node or not. Also we check the values too.